pkg/cloudprovider/cloud.go

```diff
@@ -37,6 +37,8 @@ type Interface interface {
 	Clusters() (Clusters, bool)
 	// Routes returns a routes interface along with whether the interface is supported.
 	Routes() (Routes, bool)
+	// Volumes returns a volumes interface along with whether the interface is supported.
+	Volumes() (Volumes, bool)
```

Comment:
Ok.  We'll hash out a different design during our call later. I'll either close this PR or rework it to fit what we figure out together.